scripts/main/album.js

```diff
@@ -1264,59 +1317,52 @@ album.isUploadable = function () {
 		return false;
 	}
 
-	// For special cases of no album / smart album / etc. we return true.
-	// It's only for regular non-matching albums that we return false.
-	if (album.json === null || !album.json.owner_name) {
-		return true;
-	}
-
-	return album.json.owner_name === lychee.username;
+	// TODO: Comparison of numeric user IDs (instead of names) should be more robust
+	return album.json !== null && album.json.owner_name === lychee.username;
```

Comment:
```suggestion
	return album.json === null || !album.json.owner_name || album.json.owner_name === lychee.username;
```
The removal of an `if` some 5 lines earlier resulted in a change in behavior: smart albums now appear read-only for non-admin users. This tweak restores the old behavior.